osu.Game.Rulesets.Osu/Mods/OsuModBlinds.cs

```diff
@@ -128,8 +142,21 @@ private void load()
 
             protected override void Update()
             {
-                float start = Parent.ToLocalSpace(restrictTo.ScreenSpaceDrawQuad.TopLeft).X;
-                float end = Parent.ToLocalSpace(restrictTo.ScreenSpaceDrawQuad.TopRight).X;
+                float start, end;
+
+                if (barrelRollActive)
+                {
+                    float origin = restrictTo.ToSpaceOfOtherDrawable(restrictTo.OriginPosition, Parent).X;
+                    float halfDiagonal = MathF.Sqrt(MathF.Pow(restrictTo.DrawWidth / 2, 2) + MathF.Pow(restrictTo.DrawHeight / 2, 2));
+
+                    start = origin - halfDiagonal;
+                    end = origin + halfDiagonal;
+                }
+                else
+                {
+                    start = Parent.ToLocalSpace(restrictTo.ScreenSpaceDrawQuad.TopLeft).X;
+                    end = Parent.ToLocalSpace(restrictTo.ScreenSpaceDrawQuad.TopRight).X;
+                }
```

Comment:
Is there a reason this doesn't use this new barrel roll special logic for all cases? I don't see a reason for not doing that as it seems like it would be correct for all cases, and a quick visual test on a raw blinds mod shows not much of a difference (the bg doors got a bit closer though):

`if (barrelRollActive)`/PR:

https://user-images.githubusercontent.com/22781491/126581660-6424c3ee-2b2a-4b52-b3c2-76f88389c535.mp4


`else`/master:

https://user-images.githubusercontent.com/22781491/126581656-3f69273a-e492-4ca0-8560-8b959f3fddfd.mp4

Bringing that up because I'm extremely hesitant about defining logic in a mod specific for a different mod, it has never been done and sounds pretty incorrect to do so. (for this specific case at least)

Reply:
The barrel roll logic can't be applied universally because hit circles near the horizontal sides of the playfield would appear within the inner blinds.

I'll look into a different way to make the fix work that isn't mod-specific (tracking in https://github.com/ppy/osu/pull/13971#discussion_r674471193).

Reply:
Not much point with the conversation switching but sure I guess..